### DISCUSSION TITLE
OpenType Programming/AFDKO support

### DISCUSSION THREAD

**Discussion by alephpi** [fontra/fontra#1757] [Q&A] [answered]
Hi, I just wonder if Fontra supports [OpenType Programming](https://simoncozens.github.io/fonts-and-layout/features.html), i.e. to write down the following scripts and compile and display the rendering immediately.
```
feature liga {
  sub f f by f_f;
} liga;
```
Thank you!

**Comment by alephpi** [comment, marked as answer]
Oh well, seems not yet. There's no green check after the term 'Edit OpenType features' in the roadmap.

**Reply by justvanrossum** [reply]
Yes, I'm afraid this is still on our to do list.

**Reply by alephpi** [reply]
I see. Really looking forward to it.

**Comment by mitradranirban** [comment]
While it will take some time to provide full opentype support in fontra, will it be feasible to provide a harfbuzz based opentype previewer like Fontforge is now providing https://github.com/fontforge/fontforge/pull/5522

**Reply by justvanrossum** [reply]
The plan is to use fontc to build a minimal OTF/TTF, then use harfbuzzjs to do the shaping directly in the browser. I'm afraid I cannot provide a timeline for this feature at the moment.
